packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/TimelineHeaderRow/TimelineCollapser.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+
+import { Tooltip } from 'antd';
+
+import './TimelineCollapser.css';
+import type { Span } from '../../../../types';
+
+type CollapserProps = {
+  onCollapseAll: (Span[]) => void,
+  onCollapseOne: (Span[]) => void,
+  onExpandOne: (Span[]) => void,
+  onExpandAll: () => void,
+  spans: Span[],
```

Comment:
I suggest adding the `spans` in the `TraceTimelineViewer` component, so they don't need to be passed down.